### PR TITLE
Expose a knob to let the user tune default CPU model

### PR DESCRIPTION
**Pull request by tiraboschi**
Expose a knob to let the user tune the cluster-wide
default CPU model.

Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=2096287

Signed-off-by: Simone Tiraboschi <stirabos@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Expose a knob to let the user tune default CPU model
```

### CODE REVIEW

File: controllers/operands/kubevirt_test.go [hunk 1]
```diff
@@ -1032,6 +1032,119 @@ Version: 1.2.3`)
...
 		})
 
+		Context("test CPUModel", func() {
+
+			It("should propagate the CPUModel from the HC if set", func() {
+				existKv, err := NewKubeVirt(hco)
+				Expect(err).ToNot(HaveOccurred())
+
+				testCPUModel := "testValue"
+				hco.Spec.DefaultCPUModel = &testCPUModel
```
Comment (nunnatsa) [inline controllers/operands/kubevirt_test.go:1042]: Please consider to use the `pointer` package; e.g. 
```suggestion
				hco.Spec.DefaultCPUModel = pointer.String("testValue")
```

File: controllers/operands/kubevirt_test.go [hunk 2]
```diff
@@ -1032,6 +1032,119 @@ Version: 1.2.3`)
...
+				foundResource := &kubevirtcorev1.KubeVirt{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existKv.Name, Namespace: existKv.Namespace},
+						foundResource),
+				).ToNot(HaveOccurred())
+
+				kvCPUModel := foundResource.Spec.Configuration.CPUModel
+				Expect(kvCPUModel).ToNot(BeNil())
+				Expect(kvCPUModel).To(Equal(""))
```
Comment (nunnatsa) [inline controllers/operands/kubevirt_test.go:1088]:
```suggestion
				Expect(kvCPUModel).To(BeEmpty())
```

**Review by nunnatsa — COMMENTED** [review]
Added few low priority comments in the unit tests.

File: controllers/operands/kubevirt_test.go (hunk 1 above)
Comment (tiraboschi) [reply controllers/operands/kubevirt_test.go]: done

File: controllers/operands/kubevirt_test.go (hunk 2 above)
Comment (tiraboschi) [reply controllers/operands/kubevirt_test.go]: done

File: controllers/operands/kubevirt_test.go [hunk 3]
```diff
@@ -1032,6 +1032,119 @@ Version: 1.2.3`)
...
+				foundResource := &kubevirtcorev1.KubeVirt{}
+				Expect(
+					cl.Get(context.TODO(),
+						types.NamespacedName{Name: existKv.Name, Namespace: existKv.Namespace},
+						foundResource),
+				).ToNot(HaveOccurred())
+
+				kvCPUModel := foundResource.Spec.Configuration.CPUModel
+				Expect(kvCPUModel).ToNot(BeNil())
+				Expect(kvCPUModel).To(Equal(testCPUModel))
```
[the remark this answers is not in the document]
Comment (tiraboschi) [reply controllers/operands/kubevirt_test.go:1062]: on kubevirt side it's a value

**Review by nunnatsa — APPROVED** [review]
/approve

**Comment by tiraboschi** [comment]
ignoring slightly decreased coverage
/override coverage/coveralls

**Pull request merged**